app/src/sharedTest/java/org/oppia/android/app/classroom/ClassroomListFragmentTest.kt

```diff
@@ -0,0 +1,878 @@
+   *
+   * # Logic for recommendation system
+   *
+   * We always recommend the next topic that all dependencies are completed for. If a topic with
+   * prerequisites is completed out-of-order (e.g. test topic 1 above) then we assume fractions is
+   * already done. In the same way, finishing test topic 2 means there's nothing else to recommend.
+   */
+  @Test
+  fun testFragment_markStory0DonePlayStory1FirstTestTopic_playFractionsTopic_orderIsCorrect() {
+    fakeOppiaClock.setFakeTimeMode(FakeOppiaClock.FakeTimeMode.MODE_UPTIME_MILLIS)
+    storyProgressTestHelper.markCompletedTestTopic0Story0(
+      profileId = profileId,
+      timestampOlderThanOneWeek = false
+    )
+    storyProgressTestHelper.markInProgressSavedFractionsStory0Exp0(
+      profileId = profileId,
+      timestampOlderThanOneWeek = false
+    )
+    storyProgressTestHelper.markInProgressSavedTestTopic1Story0(
+      profileId = profileId,
+      timestampOlderThanOneWeek = false
+    )
+    logIntoAdminTwice()
+
+    composeRule.onNodeWithTag(PROMOTED_STORY_LIST_HEADER_TEST_TAG).onChildAt(0)
+      .assertTextContains(context.getString(R.string.stories_for_you))
+      .assertIsDisplayed()
+
+    composeRule.onNodeWithTag(PROMOTED_STORY_LIST_TEST_TAG).apply {
+      onChildAt(0)
+        .assertTextContains("Fifth Exploration")
+        .assertTextContains("SECOND TEST TOPIC")
+        .assertTextContains("SCIENCE")
+        .assertIsDisplayed()
+
+      onChildAt(1)
+        .assertTextContains("What is a Fraction?")
+        .assertTextContains("FRACTIONS")
+        .assertTextContains("MATHS")
+        .assertIsDisplayed()
+
+      // TODO(#5344): This story should be promoted.
+      onChildAt(2)
+        .assertDoesNotExist()
+      /*.assertTextContains("What is a Ratio?")
+      .assertTextContains("RATIOS AND PROPORTIONAL REASONING")
+      .assertTextContains("MATHS")
+      .assertIsDisplayed()*/
+    }
+  }
+
+  @Test
+  // TODO(#5344): Add tests for coming soon topics list.
+  fun testFragment_markStory0OfRatiosAndTestTopics0And1Done_playTestTopicStory0_noPromotions() {
+  }
+
+  @Test
+  fun testFragment_markStory0DoneFirstTestTopic_recommendedStoriesIsCorrect() {
+    fakeOppiaClock.setFakeTimeMode(FakeOppiaClock.FakeTimeMode.MODE_UPTIME_MILLIS)
+    storyProgressTestHelper.markCompletedTestTopic0Story0(
+      profileId = profileId1,
+      timestampOlderThanOneWeek = false
+    )
+    logIntoAdminTwice()
+
+    composeRule.onNodeWithTag(PROMOTED_STORY_LIST_HEADER_TEST_TAG).onChildAt(0)
+      .assertTextContains(context.getString(R.string.recommended_stories))
+      .assertIsDisplayed()
+
+    composeRule.onNodeWithTag(PROMOTED_STORY_LIST_TEST_TAG).onChildAt(1)
+      .assertTextContains("What is a Ratio?")
+      .assertTextContains("RATIOS AND PROPORTIONAL REASONING")
+      .assertTextContains("MATHS")
+      .assertIsDisplayed()
+  }
+
+  @Test
+  @Ignore("The promoted story list only contains 'What is a Fraction?' story.")
```

Comment:
When I replicate this behaviour manually, I can recereate the test conditions and outcomes. can you tell why this is failing? 

FYI you can mack a chapter or story completed from dev options

Reply:
The test is now fixed. It was failing because a wrong `profileId` was being used.